core/src/main/java/org/infinispan/remoting/inboundhandler/BasePerCacheInboundInvocationHandler.java

```diff
@@ -52,6 +52,10 @@
    public static final String MBEAN_COMPONENT_NAME = "InboundInvocationHandler";
    private static final int NO_TOPOLOGY_COMMAND = Integer.MIN_VALUE;
 
+   // This is an approved use of blocking executor due to the fact that it only runs a task
```

Comment:
I will refactor this as mentioned in prior comment.

Reply:
Actually this one does have to stay until we fix https://issues.redhat.com/browse/ISPN-11483, since we have to invoke the `checkForReadyTasks` method. But this PR will make that one possible :)

Reply:
Is it an allowed use if it's supposed to be removed, though?

Reply:
If the comment is the thing bothering you I can remove it in this PR.... It will be removed very soon either way as I have to fix the other JIRAs that these talk about, which are my next steps.

Reply:
Well actually I won't be fixing the xsite blocking anytime soon.

Reply:
I have removed those comments so we don't have to keep talking about this :)